File: src/main/java/baboon/industry/recipe/RecipesCompressor.java
package baboon.industry.recipe;

import baboon.industry.block.I2Blocks;
import baboon.industry.item.I2Items;
import net.minecraft.core.block.Block;
import net.minecraft.core.item.Item;
import net.minecraft.core.item.ItemStack;

import java.util.HashMap;

public class RecipesCompressor {
    private static final HashMap<Integer, ItemStack> recipeList = new HashMap<>();

    public RecipesCompressor() {
        addRecipe(I2Items.ingotTin.id, new ItemStack(I2Items.plateTin, 2));
        addRecipe(I2Blocks.blockTin.id, new ItemStack(I2Items.plateTin, 18));

        addRecipe(I2Items.ingotCopper.id, new ItemStack(I2Items.plateCopper, 2));
        addRecipe(I2Blocks.blockCopper.id, new ItemStack(I2Items.plateCopper, 18));

        addRecipe(I2Items.ingotBronze.id, new ItemStack(I2Items.plateBronze, 2));
        addRecipe(I2Blocks.blockBronze.id, new ItemStack(I2Items.plateBronze, 18));

        addRecipe(I2Items.ingotIridium.id, new ItemStack(I2Items.plateIridium, 1));

        addRecipe(Item.ingotIron.id, new ItemStack(I2Items.plateIron, 2));
        addRecipe(Block.blockIron.id, new ItemStack(I2Items.plateIron, 18));

        addRecipe(Item.ingotGold.id, new ItemStack(I2Items.plateGold, 2));
        addRecipe(Block.blockGold.id, new ItemStack(I2Items.plateGold, 18));

        addRecipe(Item.ingotSteel.id, new ItemStack(I2Items.plateSteel, 2));
        addRecipe(Block.blockSteel.id, new ItemStack(I2Items.plateSteel, 18));

        addRecipe(I2Blocks.hardenedCoal.id, new ItemStack(Item.diamond, 1));
    }

    public static void addRecipe(int input, ItemStack output) {
        recipeList.put(input, output);
    }

    public ItemStack getRecipeResult(int i) {
        return recipeList.get(i);
    }

    public HashMap<Integer, ItemStack> getRecipeList() {
        return recipeList;
    }
}

File: src/main/kotlin/turniplabs/industry/recipes/RecipesCompressor.kt
package turniplabs.industry.recipes

import net.minecraft.core.block.Block
import net.minecraft.core.item.Item
import net.minecraft.core.item.ItemStack
import turniplabs.industry.blocks.IndustryBlocks
import turniplabs.industry.items.IndustryItems

object RecipesCompressor {
    private val recipeList = HashMap<Any?, Any?>()

    init {
        addRecipe(IndustryItems.copperIngot, IndustryItems.copperPlate, 2)
        addRecipe(IndustryItems.tinIngot, IndustryItems.tinPlate, 2)
        addRecipe(IndustryItems.bronzeIngot, IndustryItems.bronzePlate, 2)
        addRecipe(Item.ingotIron, IndustryItems.ironPlate, 2)
        addRecipe(Item.ingotGold, IndustryItems.goldPlate, 2)
        addRecipe(Item.ingotSteel, IndustryItems.steelPlate, 2)
        addRecipe(IndustryItems.rawUranium, IndustryItems.uraniumIngot, 1)

        addRecipe(IndustryBlocks.copperBlock, IndustryItems.copperPlate, 18)
        addRecipe(IndustryBlocks.tinBlock, IndustryItems.tinPlate, 18)
        addRecipe(IndustryBlocks.bronzeBlock, IndustryItems.bronzePlate, 18)
        addRecipe(Block.blockIron, IndustryItems.ironPlate, 18)
        addRecipe(Block.blockGold, IndustryItems.goldPlate, 18)
        addRecipe(Block.blockSteel, IndustryItems.steelPlate, 18)

        addRecipe(IndustryBlocks.hardenedCoal, Item.diamond, 1)
    }

    fun addRecipe(itemInput: Item, outputItem: Item, outputCount: Int) {
        recipeList[itemInput.id] = ItemStack(outputItem, outputCount)
    }

    fun addRecipe(itemInput: Item, outputBlock: Block, outputCount: Int) {
        recipeList[itemInput.id] = ItemStack(outputBlock, outputCount)
    }

    fun addRecipe(blockInput: Block, outputItem: Item, outputCount: Int) {
        recipeList[blockInput.id] = ItemStack(outputItem, outputCount)
    }

    fun addRecipe(blockInput: Block, outputBlock: Block, outputCount: Int) {
        recipeList[blockInput.id] = ItemStack(outputBlock, outputCount)
    }

    fun getResult(i: Int): ItemStack? {
        return recipeList[i] as ItemStack?
    }

    fun getRecipeList(): HashMap<Any?, Any?> {
        return recipeList
    }
}